src/NServiceBus.Core/Transports/Msmq/QueueCreator.cs

```diff
@@ -30,36 +30,59 @@ public Task CreateQueueIfNecessary(QueueBindings queueBindings, string identity)
 
         void CreateQueueIfNecessary(string address, string identity)
         {
-            if (address == null)
+            var msmqAddress = MsmqAddress.Parse(address);
+
+            Logger.Debug($"Creating '{address}' if needed.");
+
+            if (msmqAddress.IsRemote)
             {
+                Logger.Info($"'{address}' is a remote queue and won't be created");
                 return;
             }
 
-            var msmqAddress = MsmqAddress.Parse(address);
+            var queuePath = msmqAddress.PathWithoutPrefix;
 
-            Logger.Debug($"Creating '{address}' if needed.");
+            if (MessageQueue.Exists(queuePath))
+            {
+                Logger.Debug($"'{address}' already exists");
+                return;
+            }
 
-            MessageQueue queue;
-            if (MsmqUtilities.TryOpenQueue(msmqAddress, out queue) || MsmqUtilities.TryCreateQueue(msmqAddress, identity, settings.UseTransactionalQueues, out queue))
+            try
             {
-                using (queue)
+                using (var queue = MessageQueue.Create(queuePath, useTransactionalQueues))
                 {
-                    Logger.Debug("Setting queue permissions.");
+                    Logger.Debug($"Created queue, path: [{queuePath}], identity: [{identity}], transactional: [{useTransactionalQueues}]");
 
                     try
                     {
-                        QueuePermissions.SetPermissionsForQueue(queue, identity);
+                        queue.SetPermissions(identity, MessageQueueAccessRights.WriteMessage);
+                        queue.SetPermissions(identity, MessageQueueAccessRights.ReceiveMessage);
+                        queue.SetPermissions(identity, MessageQueueAccessRights.PeekMessage);
+
+                        queue.SetPermissions(LocalAdministratorsGroupName, MessageQueueAccessRights.FullControl);
+
+                        // Everyone & Anonymous is granted write message permissions on all Windows / Windows Server versions after 2003.
```

Comment:
Remove this code since 2003 is no longer supported? See https://msdn.microsoft.com/en-us/library/bb822049(v=vs.110).aspx since 4.5.1 is not supported on 2003 anyway